server/models/services/TestPlanTargetService.js

```diff
@@ -61,25 +63,47 @@ const createTestPlanTarget = async (
     { title, at, atVersion, browser, browserVersion },
     testPlanTargetAttributes = TEST_PLAN_TARGET_ATTRIBUTES
 ) => {
-    // TODO: Construct title as <at> <atVersion> with <browser> <browserVersion>
+    let atName;
+    let browserName;
 
-    // TODO: Check if at, atVersion, browser, browserVersion exists and create it not on the fly? Or assume they exist?
+    if (typeof at === 'string') {
+        atName = at;
+        const atResults = await AtService.getAts(null, { name: at });
+        at = atResults.length ? atResults[0].id : 0;
+    } else {
+        // assuming type is number; get name to construct title if required
+        const atResult = await AtService.getAtById(at);
+        atName = atResult ? atResult.name : '';
+    }
+
+    if (typeof browser === 'string') {
+        browserName = browser;
+        const browserResults = await BrowserService.getBrowsers(null, {
+            name: browser
+        });
+        browser = browserResults.length ? browserResults[0].id : 0;
+    } else {
+        // assuming type is number; get name to construct title if required
+        const browserResult = await BrowserService.getBrowserById(browser);
+        browserName = browserResult ? browserResult.name : '';
+    }
+
+    // Construct title as <at> <atVersion> with <browser> <browserVersion>
+    if (!title)
+        title = `${atName} ${atVersion} with ${browserName} ${browserVersion}`;
```

Comment:
I actually think this attribute makes more sense to be in GraphQL only. However, since I'm about to migrate the data, I think it's easier to leave it in for now and let me do it in a day or two!

Let me know if you agree with that?

Reply:
So I think it's better managed here just in case there is an instance where a custom title is required to be passed or having the title manipulated in some other way (which this supports) and having that persisted. so I don't think that should necessarily be dependent on the GraphQL query. I could see instances where definitely having a saved `title` would be ideal and also cases against but I'm definitely up for discussing it further

Reply:
Having a "defaulted" title persisted on the model and the GraphQL having it's own variation based on something specific for its query works as well.

Reply:
Sure, let's keep it then!